### PR TITLE
Stabilize CI-build-and-test jobs on Ubuntu Linux 22.04

### PR DESCRIPTION
**Pull request by gapisback**
This commit attempts to keep the CI tests runing cleanly. 

Some tests  are failing due to different issues reported under issue #242 and #244. 

Those will need some minor corrections in build-steps.

For now, 5 test-cases from test.sh are commented out with a  Warning message.

 - Skip few simulated-SEV related build-and-test cases.
 - Skip ISLET-related build-and-test tests due to rustc version incompatibility.

Those will need to be re-enabled when above two issues are resolved, if possible.

(If further investigation shows that we cannot support Ubunutu 22.04, then some changes will need to be made in CI's build.yml to drop-down  to an older Ubuntu rev-level.)

### CODE REVIEW

File: sample_apps/run_example.sh [hunk 1]
```diff
@@ -681,7 +681,9 @@ function rm_non_git_files() {
     fi
 
     local tmp_islet_dir="/tmp/islet-save"
-    mkdir "${tmp_islet_dir}"
+    if [ ! -d "${tmp_islet_dir}" ]; then
+        mkdir "${tmp_islet_dir}"
```
Comment (gapisback) [inline sample_apps/run_example.sh:685]: Minor unrelated fix. Re-runs of this script will fail if this dir exists.

File: sev-snp-simulator/Makefile [hunk 2]
```diff
@@ -48,7 +48,7 @@ clean:
 	rm -f modules.order Module.symvers Module.markers *.o *.cmd sevnull.ko
 	rm -rf keys
 
-.PHNOY:
+.PHONY:
```
Comment (gapisback) [inline sev-snp-simulator/Makefile:51]: Fix typo.

**Review by yelvmw — APPROVED** [review]
LGTM. Thanks!
